homeassistant/components/climate/tuya.py

```diff
@@ -0,0 +1,166 @@
+    STATE_PERFORMANCE: 'performance',
+}
+
+TUYA_STATE_TO_HA = {value: key for key, value in HA_STATE_TO_TUYA.items()}
+
+FAN_MODES = {SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH}
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up Tuya Climate devices."""
+    if discovery_info is None:
+        return
+    tuya = hass.data[DATA_TUYA]
+    dev_ids = discovery_info.get('dev_ids')
+    devices = []
+    for dev_id in dev_ids:
+        device = tuya.get_device_by_id(dev_id)
+        if device is None:
+            continue
+        devices.append(TuyaClimateDevice(device))
+    add_devices(devices)
+
+
+class TuyaClimateDevice(TuyaDevice, ClimateDevice):
+    """Tuya climate devices,include air conditioner,heater."""
+
+    def __init__(self, tuya):
+        """Init climate device."""
+        super().__init__(tuya)
+        self.entity_id = ENTITY_ID_FORMAT.format(tuya.object_id())
+
+    @property
+    def is_on(self):
+        """Return true if climate is on."""
+        return self.tuya.state()
+
+    @property
+    def precision(self):
+        """Return the precision of the system."""
+        return PRECISION_WHOLE
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement used by the platform."""
+        unit = self.tuya.temperature_unit()
+        if unit == 'CELSIUS':
+            return TEMP_CELSIUS
+        elif unit == 'FAHRENHEIT':
+            return TEMP_FAHRENHEIT
+        return TEMP_CELSIUS
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        mode = self.tuya.current_operation()
+        if mode is None:
+            return None
+        return TUYA_STATE_TO_HA.get(mode)
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        modes = self.tuya.operation_list()
+        operation_list = []
+        for mode in modes:
+            operation_list.append(TUYA_STATE_TO_HA.get(mode))
+        return operation_list
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self.tuya.current_temperature()
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self.tuya.target_temperature()
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return self.tuya.target_temperature_step()
+
+    @property
+    def current_fan_mode(self):
+        """Return the fan setting."""
+        return self.tuya.current_fan_mode()
+
+    @property
+    def fan_list(self):
+        """Return the list of available fan modes."""
+        return self.tuya.fan_list()
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        self.tuya.set_temperature(temperature)
```

Comment:
Probably guard for `None` value before setting the temperature. The temperature argument is optional in the service schema för this method.